scico/__init__.py

```diff
@@ -12,6 +12,27 @@
 
 import sys
 
+# Python 3.7 compatibility
```

Comment:
Instead of the `try/except` dance, perhaps it would be cleaner to just directly check `sys.version_info`?

Also, perhaps put all the 3.7 compatibility stuff into a separate file (e.g. `_py37.py`) to be imported by `__init__.py`?